functions/gravityForms/getGfFieldValidationSchema.js

```diff
@@ -1,39 +1,116 @@
 import getGfFieldId from '@/functions/gravityForms/getGfFieldId'
 import * as Yup from 'yup'
 
-// TODO Refactor this file for scaling purposes.
-
 /**
- * Map props to validation schemas.
+ * Create Yup string schema object.
  *
- * @param {Object} fieldData GravityForm field props.
+ * @author Mike England <mike.england@webdevstudios.com>
+ * @since 2021-01-07
+ * @link https://github.com/jquense/yup#string
  */
-export default function getGfFieldValidationSchema(fieldData) {
-  const validationTypes = {
-    text: 'string'
+class StringSchemaFactory {
+  /**
+   * Constructor.
+   *
+   * Note: wp-graphql-gravity-forms plugin may need to be replaced.
+   *
+   * @param {Object} fieldData from GravityForm GraphQL data Object.
+   * @link https://github.com/harness-software/wp-graphql-gravity-forms
+   */
+  constructor(fieldData) {
+    this.fieldData = fieldData
   }
 
-  const type = validationTypes?.[fieldData?.type]
-
-  if (!type) {
-    return
+  /**
+   * Get Yup schema object with validation requirements.
+   *
+   * Combine multiple schemas into one.
+   *
+   * @author Mike England <mike.england@webdevstudios.com>
+   * @since 2021-01-07
+   * @returns {Object} Combined Yup validationSchema Object.
+   * @link https://github.com/jquense/yup#mixedconcatschema-schema-schema
+   */
+  get schema() {
+    return Yup.string()
+      .concat(this.getMaxLengthSchema())
+      .concat(this.getRequiredSchema())
   }
 
-  const yupType = Yup[type]()
+  /**
+   * Get Yup required field validaion.
+   *
+   * @author Mike England <mike.england@webdevstudios.com>
+   * @since 2021-01-07
+   * @return {Object} Yup validationSchema Object.
+   * @link https://github.com/jquense/yup#stringrequiredmessage-string--function-schema
+   */
+  getRequiredSchema() {
+    if (!this.fieldData?.isRequired) {
+      return
```

Comment:
Does this work as expected with just `return` (thus `undefined`) or would `return null` be better?

Reply:
It does work without issue, I was curious about the same thing because I had that initially. When `concat()` without a value nothing happens.